packages/headless/src/features/instant-results/instant-results-actions.ts

```diff
@@ -0,0 +1,117 @@
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {InstantResultSection} from '../../state/state-sections';
+import {
+  validatePayload,
+  requiredEmptyAllowedString,
+  requiredNonEmptyString,
+} from '../../utils/validate-payload';
+import {
+  isErrorResponse,
+  AsyncThunkSearchOptions,
+} from '../../api/search/search-api-client';
+import {StateNeededByExecuteSearch} from '../search/search-actions';
+import {mapSearchRequest, mapSearchResponse} from '../search/search-mappings';
+import {buildSearchAndFoldingLoadCollectionRequest} from '../search-and-folding/search-and-folding-request';
+import {logQueryError} from '../search/search-analytics-actions';
+import {Result} from '../../case-assist.index';
+
+export interface RegisterInstantResultActionCreatorPayload {
+  /**
+   * The search box id.
+   */
+  id: string;
+}
+
+export interface UpdateInstantResultQueryActionCreatorPayload {
+  /**
+   * The search box id.
+   */
+  id: string;
+  /**
+   * The initial basic query expression for instant results.
+   */
+  q: string;
+}
+
+const instantResultsRegisterDefinition = {
+  id: requiredNonEmptyString,
+};
+
+const instantResultsQueryDefinition = {
+  ...instantResultsRegisterDefinition,
+  q: requiredEmptyAllowedString,
+};
+
+export const registerInstantResults = createAction(
+  'instantResults/register',
+  (payload: RegisterInstantResultActionCreatorPayload) =>
+    validatePayload(payload, instantResultsRegisterDefinition)
+);
+
+export const updateInstantResultsQuery = createAction(
+  'instantResults/updateQuery',
+  (payload: UpdateInstantResultQueryActionCreatorPayload) =>
+    validatePayload(payload, instantResultsQueryDefinition)
+);
+
+export interface FetchInstantResultsActionCreatorPayload {
+  /**
+   * The search box id.
+   */
+  id: string;
+  /**
+   * The query for which instant results are retrieved.
+   */
+  q: string;
+}
+
+export interface FetchInstantResultsThunkReturn {
+  results: Result[];
+}
+
+export type StateNeededByInstantResults = StateNeededByExecuteSearch &
+  InstantResultSection;
+
+export const fetchInstantResults = createAsyncThunk<
```

Comment:
At a practical level, the difference between this and the classic "executeSearch" action in `search-actions.ts` is very subtle. Should the automatic did you mean feature kick in here ? 🤔 

Maybe we should move that action there instead ? So they are all co-located, and can share more code with one another.

We most likely also want to log an analytics event associated with it. This can come in a later PR.